docs/python/generators/doxygen.py

```
@@ -382,35 +470,23 @@ def _example_priority(e):
                     print(f"??? note \"{src_rel}\"", file=fos)
-                    print(f"    [{ex['title']}]({ex['id']}.md) - {ex.get('description','')}", file=fos)
-                    print(f"    ```{extension}{hl_attr}", file=fos)
-                    for line in ex['snippet'].splitlines():
-                        print(f"    {line}", file=fos)
-                    print(f"    ```", file=fos)
+                    print(f"    {ex['title']}{f' - {ex.get("description", "")}' if ex.get('description', '') else ''}", file=fos)
+                    if snippet:
+                        # Strip leading blank lines so hl_lines numbering matches compute_hl_lines
+                        lines = snippet.splitlines()
+                        first_nonblank = 0
+                        for l in lines:
```

Ruff (0.15.21)

[error] Ambiguous variable name: `l`

(E741)